lib/matplotlib/widgets.py

```diff
@@ -2699,9 +2699,8 @@ def _press(self, event):
             self.update()
 
         if self._active_handle is None:
-            x = event.xdata
-            y = event.ydata
-            self.visible = False
+            x = self._eventpress.xdata
+            y = self._eventpress.ydata
```

Comment:
Please keep this using `event`; I might remove `_eventpress` as a state.

Reply:
Removing `_eventpress` might be a little difficult, as `_release(event)` refers to `_eventpress` to calculate span, and also for calling `onselect`. The reference for the span calculation might be able to be gotten around, but we'd then need to reconstruct the press event to send to `onselect`.

Reply:
I'm aware, but at this point `event == self._eventpress`, so it should be fine to go back.

Reply:
Done. Also updated `_release` to refer to `event` instead of `_eventrelease`. `_release` still needs `_eventpress` to properly calculate the span.